### PR TITLE
🐛 bug fix:fix issue in entitypage tabs with replacing nav link tabs

### LINKED ISSUE #19912: 🐛 Bug Report: Tabs that change the active route (such as EntityPage tabs) should use a link element
### 📜 Description

Tabs that change routes use the incorrect element for display,

### 👍 Expected behavior

They should instead use links like demonstrated here: https://mui.com/material-ui/react-tabs/#nav-tabs

Because the current element is a button with an onClick handler it is impossible to copy link or open in new browser tab. This _may_ fail some a11y s tandards.

### 👎 Actual Behavior with Screenshots

Tabs that change routes such as the "routable extensions" that show up in the EntityPage use a button element from which you cannot copy a link or open in a new browser tab.

### 👟 Reproduction steps

1. Go to an EntityPage in the catalog
2. Try to middle click or right click any of the tabs

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by npiyush97**
## Hey, I just made a Pull Request!
fix #19912

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

prev
Screenshot from 2024-03-14 11-49-32
after 
Screenshot from 2024-03-14 11-48-30

### CODE REVIEW

**Comment by npiyush97** [comment]
just to get the maintainers' view 
clean up required this is not the final commit.

File: packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx [hunk 1]
```diff
@@ -17,6 +17,7 @@ import Box from '@material-ui/core/Box';
 import { makeStyles } from '@material-ui/core/styles';
 import TabUI, { TabProps } from '@material-ui/core/Tab';
 import Tabs from '@material-ui/core/Tabs';
+import Tab from '@material-ui/core/Tab';
```
Comment (tudi2d) [inline packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx:20]: You are importing the same component here twice. `Tab` & `TabUI` both refer to the `default export` of `'@material-ui/core/Tab'`.

File: packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx [hunk 2]
```diff
@@ -57,18 +58,42 @@ const useStyles = makeStyles(
...
   label: string;
   tabProps?: TabProps<React.ElementType, { component?: React.ElementType }>;
 };
 
 type HeaderTabsProps = {
-  tabs: Tab[];
+  tabs: TabProp[];
   onChange?: (index: number) => void;
   selectedIndex?: number;
 };
```
Comment (tudi2d) [inline packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx:71]: Let's keep this renaming out of this PR for simplicity - no real value in calling it `TabProps` as there is no exported `Tab ` component from this file.

Comment (tudi2d) [reply packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx:71]: checkout `yarn tsc` to see current issue with renaming

File: packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx [hunk 3]
```diff
@@ -108,14 +133,16 @@ export function HeaderTabs(props: HeaderTabsProps) {
         value={selectedTab}
       >
         {tabs.map((tab, index) => (
-          <TabUI
+          <LinkTab
             {...tab.tabProps}
+            href={window.location.href}
```
Comment (tudi2d) [inline packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx:138]: I don't think `window.location.href` is what you are looking for here.

**Review by tudi2d — CHANGES_REQUESTED** [review]
A few comment's - not super sure this is the best approach. At the moment the workaround with the type is possible for MUI `v4`, but it would change in `v5` due to the removal of the `href` prop from the `BaseButton`. 

As we have a thrid-party router we probably want something like this: https://mui.com/material-ui/integrations/routing/#tabs with our own `Link` component, rather than hardcoding a `href`. This way we would get the benefits of `react-router`.

File: packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx (hunk 3 above)
Comment (npiyush97) [reply packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx:138]: how to pass the path then ?
```
import Link from 'backstage/component'
<tabui 
component={Link}
/>
```
if i pass to = {tab.id} it doesn't append path after `catalog/tab.id` path it directs to `localhost/tab.id` 
I have some ideas but any suggestions?

**Comment by npiyush97** [comment]
@tudi2d thought on changes? help with tests?

File: packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx [hunk 4]
```diff
@@ -95,7 +96,11 @@ export function HeaderTabs(props: HeaderTabsProps) {
       setSelectedTab(selectedIndex);
     }
   }, [selectedIndex]);
-
+  function removeLeadingSlash(path: string) {
+    let newPath = path;
+    newPath = path.replace(/^\//, '');
+    return newPath;
+  }
```
Comment (benjdlambert) [inline packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx:103]:
```suggestion
  function removeLeadingSlash(path: string) {
    return path.replace(/^\//, '');
  }
```

**Review by benjdlambert — CHANGES_REQUESTED** [review]
think we can simplify the function a little bit

File: packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx [hunk 5]
```diff
@@ -109,13 +112,15 @@ export function HeaderTabs(props: HeaderTabsProps) {
       >
         {tabs.map((tab, index) => (
           <TabUI
-            {...tab.tabProps}
             data-testid={`header-tab-${index}`}
             label={tab.label}
             key={tab.id}
+            component={Link}
+            to={removeLeadingSlash(tab.id)}
```
Comment (tudi2d) [inline packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx:119]: are you sure the `removeLeadingSlash` is required here & nothing that can be fixed by the `relative`  prop of the React Router `Link`: https://reactrouter.com/en/main/components/link#relative

**Review by tudi2d — APPROVED** [review]
almost there I believe. just want to check again if the leading slash really has to be removed - haven't found good evidence for removing/not removing :/

File: packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx (hunk 5 above)
Comment (npiyush97) [reply packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx:119]: A relative <Link to> value (that does not begin with /) resolves relative to the parent route, which means that it builds upon the URL path that was matched by the route that rendered that <Link>.
for ex tab.id = '/ci-cd'  just removing slash

Comment (tudi2d) [reply packages/core-components/src/layout/HeaderTabs/HeaderTabs.tsx:119]: yes. that's why `relative="path"` might work here instead. anyways - I approved & happy to go with whatever:)

**Comment by benjdlambert** [comment]
@npiyush97 can you add a test for this please? :pray:

**Comment by npiyush97** [comment]
👉 @benjdlambert

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**